### ISSUE TITLE
Unexpected parsing error with trailing comma in destructuring pattern

### ISSUE THREAD

**Issue by sjdemartini** [eslint/eslint#5756]
**What version of ESLint are you using?**
v2.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
`"ecmaVersion": 7,`

**Please show your full configuration:**
```
  "extends": ["airbnb"],
  "plugins": [
    "react",
    "import"  
  ],
  "env": {
    "browser": true,
    "node": true,
  },
  "parserOptions": {
    "ecmaVersion": 7,
  },
  "ecmaFeatures": {
    "arrowFunctions": true,
    "blockBindings": true,
    "classes": true,
    "defaultParams": true,
    "destructuring": true,
    "forOf": true,
    "generators": false,
    "modules": true,
    "objectLiteralComputedProperties": true,
    "objectLiteralDuplicateProperties": false,
    "objectLiteralShorthandMethods": true,
    "objectLiteralShorthandProperties": true,
    "restParams": true,
    "spread": true,
    "superInFunctions": true,
    "templateStrings": true,
    "jsx": true
  },
```

**What did you do? Please include the actual source code causing the issue.**
```
  return Promise.all([
    getIsFoo(),
    getIsBar(),
    getIsBaz(),
  ]).then(([
    isFoo,
    isBar,
    isBaz,
  ]) => (
    isFoo && isBar && isBaz
  ));
```

**What did you expect to happen?**
No linting errors.

**What actually happened? Please include the actual, raw output from ESLint.**
`Error - Parsing error: Trailing comma is not permitted in destructuring patterns`

From what I understand, the trailing comma _should_ be permitted in destructuring patterns. This seems like a bug. When adding the trailing comma, I then got the conflicting ``Error - Missing trailing comma. (comma-dangle)` (due to my eslint config), so there was no way to satisfy this condition. But more importantly, the trailing comma with destructuring should have been allowed.

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed this on the online demo.

Surely, `ArrayBindingPattern` allows trailing comma ([spec](https://tc39.github.io/ecma262/2016/#prod-ArrayBindingPattern)).
Acorn seems to throw this syntax error only in the parameters of arrow functions.
@michaelficarra Do you have any insights?


![image](https://cloud.githubusercontent.com/assets/1937871/14223195/215693e0-f8b6-11e5-84d3-66960ce1a584.png)

**Comment by michaelficarra** [comment]
@mysticatea Acorn has a bug.

**Comment by mysticatea** [comment]
This has been fixed in Acorn: https://github.com/ternjs/acorn/issues/400#event-630129428
I'm waiting for the next release.

**Comment by mysticatea** [comment]
Acorn v3.1.0 has been released.
But some espree's tests are failed with that.
I'm working on espree.
